### PR TITLE
Implementing Source and Time filters

### LINKED ISSUE #20: Remove anchor links in README
Now that we're on GitHub we don't need to manually create anchors, they're provided for us!

### PR DESCRIPTION
**Pull request by venky526**
We implemented variable filters before. This is to implement the rest of the available filters Source and Time. Here's an example:
```python
from signal_analog.flow import Data
from signal_analog.charts import TimeSeriesChart
from signal_analog.dashboards import Dashboard
from signal_analog.filters import DashboardFilters, FilterVariable, FilterSource, FilterTime

program = Data('cpu.utilization').publish()
chart = TimeSeriesChart().with_name('Sample Chart Name').with_program(program)

dashboard = Dashboard().with_name('Dashboard With Filters').with_charts(chart)

aws_src = FilterSource().with_property("aws_region").with_value('us-west-2')

time = FilterTime().with_start("-1h").with_end("Now")

app_filter = DashboardFilters() \
    .with_sources(aws_src) \
    .with_time(time)

dashboard_with_filters = dashboard.with_filters(app_filter).with_api_token(api_token).update()

print(dashboard_with_filters)

```

**Note:** This PR also fixes #20 and #25

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -253,10 +246,18 @@ env_var = FilterVariable().with_alias('env')\
 .with_is_required(True)\
 .with_value('prod')
 
+aws_src = FilterSource().with_property("aws_region").with_value('us-west-2')
+
+time = FilterTime().with_start("-1h").with_end("Now")
```
Comment (dogonthehorizon) [inline README.md]: I'm wondering if it makes sense to provide a constant for this. Do we know if SignalFx treats this value as case-insensitive? Just trying to think of ways to reduce user error here haha.

File: README.md [hunk 2]
```diff
@@ -337,47 +338,6 @@ detector.with_api_token('foo').create()
 To add multiple alerting rules we would need to use different `detect`
 statements with distinct `label`s to differentiate them from one another.
 
-#### Detectors that Combine Data Streams
```
Comment (dogonthehorizon) [inline README.md]: Please pull in changes from master!

File: README.md (hunk 1 above)
Comment (venky526) [reply README.md]: It doesn't allow upper case unit indicators. We are already checking that and raising an error if it's not a lower case as well
```python
                    elif st[-1:] not in ['m', 'h', 'd', 'w']:
                        raise ValueError(
                            "Start time value must end with a units indicator. Allowed units indicators are "
                            "m, h, d, w representing minutes, hours, days, and weeks respectively. Instead, "
                            "got '{0}'".format(st[-1:]))

```

File: README.md (hunk 2 above)
Comment (venky526) [reply README.md]: I actually created a branch from master yesterday before working on this. Are you seeing any discrepancies?

File: README.md (hunk 1 above)
Comment (venky526) [reply README.md]: Also, start and end times can be either Strings or Integers. So, had to implement this way

File: README.md (hunk 2 above)
Comment (dogonthehorizon) [reply README.md]: Well this entire section shouldn't be removed :P You can see it in [master here](https://github.com/Nike-Inc/signal_analog#detectors-that-combine-data-streams). Maybe your master was behind somehow?

File: README.md (hunk 1 above)
Comment (dogonthehorizon) [reply README.md]: Makes sense. Does that also apply for `Now` vs `now`?

Comment (venky526) [reply README.md]: Yes. But, I was converting it to title case and checking which I shouldn't. Fixed now

File: README.md (hunk 2 above)
Comment (venky526) [reply README.md]: For some reason, I am not able to pull that from the master 😞  Added it manually

**Pull request merged**